Add Collabora Online

xzfc opened the pull request:
## Description of changes

URL: https://www.collaboraonline.com

This PR adds Collabora Online Development Edition (CODE). CODE is a way to run LibreOffice in the browser. Continuation of #329525.

In this PR two things are introduced:
- A package `collabora-online`.
- A NixOS module to deploy it.

cc: @7c6f434c

fixes #333457

## How to run

This guide will let you run a NixOS VM with Collabora Online and Nextcloud. The purpose is to provide a quick start and to demonstrate that it works, so it's messy and insecure.

Note: I am still yet to figure out how to properly run host-accessible NixOS VMs[^1], and not port-forwarded ones. That's why the hostname of the host and the VM should match.

1. **Build VM**
   Grab this definition of NixOS VM and replace `myawesomehostname` with the hostname of your host. (the VM and the host should have the same hostname)

   <details>
   <summary><code>vm.nix</code></summary>

   ```nix
   {
     myawesomehostname =
       { config, pkgs, ... }:
       {
         services.nextcloud = {
           enable = true;
           hostName = "myawesomehostname";
           config.adminpassFile = "/etc/nextcloud-admin-pass";
           extraApps = {
             inherit (config.services.nextcloud.package.packages.apps) richdocuments;
           };
         };
         environment.etc."nextcloud-admin-pass".text = "a";
   
         services.collabora-online = {
           enable = true;
           settings = {
             ssl.enable = false;
             ssl.termination = false;
           };
           aliasGroups = [ { host = "http://myawesomehostname"; } ];
         };
   
         systemd.services.nextcloud-setup-collabora = {
           after = [ "nextcloud-setup.service" ];
           wantedBy = [ "multi-user.target" ];
           path = [ config.services.nextcloud.occ ];
           serviceConfig = {
             Type = "oneshot";
             User = "nextcloud";
             ExecStart = pkgs.writeScript "nextcloud-setup-collabora.sh" ''
               #!${pkgs.runtimeShell}
               nextcloud-occ config:app:set richdocuments disable_certificate_verification --value yes
               nextcloud-occ richdocuments:setup --wopi-url=http://myawesomehostname:9980
             '';
           };
         };
   
         networking.firewall.enable = false;
         virtualisation = {
           forwardPorts = [
             {
               from = "host";
               guest.port = 80;
               host.port = 80;
             }
             {
               from = "host";
               guest.port = 9980;
               host.port = 9980;
             }
           ];
         };
   
         system.stateVersion = "24.05";
       };
   }
   ```

   </details>

   Build it with the following command:
   ```sh
   nix-build \
       -I nixpkgs=/path/to/nixpkgs \
       "<nixpkgs/nixos/modules/installer/tools/nixos-build-vms/build-vms.nix>" \
       --argstr networkExpr "$PWD/vm.nix"
   ```

2. **Run VM**

   ```sh
   sudo ./result/bin/nixos-run-vms
   ```

   Sudo is required to bind the VM to port 80.

3. **Login**
   In a browser, go to `http://myawesomehostname`.
   It will prompt you with a login page. Enter `root` and `a`.

4. **Create a new document**
   In the top left corner select "Files", then "+ New", then "New Document". Press "Create".
   Click on the newly created document to open it.
   The Collabora Online editor should appear.


[^1]: They should use qemu's `-net tap`, I guess.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

xzfc committed:
collabora-online: init at 24.04.6-1

xzfc committed:
nixos/collabora-online: init

SuperSandro2000 commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+
+  nativeBuildInputs = [
+    autoreconfHook
+    libcap
+    nodejs
+    npmHooks.npmConfigHook
+    pkg-config
+    python3
+    python3.pkgs.lxml
+    python3.pkgs.polib
+    rsync
```
libcap goes to buildInputs

xzfc replied:
Removed as it is not required after a bump to `24.04.6-1`.

SuperSandro2000 commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+    nodejs
+    npmHooks.npmConfigHook
+    pkg-config
+    python3
+    python3.pkgs.lxml
+    python3.pkgs.polib
+    rsync
```
We should probably build a python env with the modules

xzfc replied:
Why? This pattern is widely used in nixpkgs: https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+"++python3.pkgs."&type=code
These only used during the build, not in runtime.

SuperSandro2000 commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+  postPatch =
+    ''
+      cp ${./package-lock.json} ${finalAttrs.npmRoot}/package-lock.json
+      patchShebangs .
```
Please be a bit more specific than everything. Also we normally place postPatch right after patches and src

xzfc replied:
Updated.

SuperSandro2000 commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+    + ''
+      substituteInPlace common/JailUtil.cpp --replace-fail \
+        'Poco::Path(Util::getApplicationPath(), "coolmount").toString()' \
+        'std::string("/run/wrappers/bin/coolmount")'
+      substituteInPlace wsd/COOLWSD.cpp --replace-fail \
+        ' forKitPath += "coolforkit";' \
+        ' forKitPath = "/run/wrappers/bin/coolforkit";'
```
If we take them from PATH, then the package easier works on other distros

xzfc replied:
The code assumes absolute paths. I could come up with a patch that calls `which coolforkit` from C++ code, but it will be not exactly a trivial patch. Are we okay with such patches?

Or maybe something like accepting a `NIX_COLLABORA_COOLFORKIT` env var. Do we have anything like these in nixpkgs?

SuperSandro2000 replied:
If the code wants absolute paths, then we hardcode them for now.

Changing c/++ code is always open for memory issues and it is usually good to avoid that if possible.

We have such envs in nixpkgs but only if it needs to be dynamic. Lets start simple here first.

xzfc replied:
Removed these fixups entirely as they are not required with `mount_namespaces` option.

SuperSandro2000 commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+        'Poco::RegularExpression gitHashRe("/([0-9a-f.]+)/");' \
+    '';
+
+  postInstall = ''
+    cp etc/ca-chain.cert.pem etc/cert.pem etc/key.pem $out/etc/coolwsd
+  '';
```
Don't we want to take ca-certs?

xzfc replied:
WDYM? These self-signed certificates are provided for testing purposes.
In NixOS module, they're enabled by default unless overridden by the user.

SuperSandro2000 commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+    src = null;
+    # TODO: Use upstream `npm-shrinkwrap.json` once it's fixed
+    # https://github.com/CollaboraOnline/online/issues/9644
+    postPatch = ''cp ${./package-lock.json} package-lock.json'';
```
```suggestion
    postPatch = ''
      cp ${./package-lock.json} package-lock.json
    '';
```

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+          throw "Unsupported type: ${builtins.toJSON x}";
+    in
+    # Example config file is taken from https://github.com/CollaboraOnline/online/blob/master/coolwsd.xml.in
+    pkgs.writeText "coolwsd.xml" ''
```
How about we take the vendored file, import it from the package and then use an xml manipulation tool to substitute the values in?

xzfc replied:
Rehauled in the RFC42-style.

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+          "pt_PT"
+          "ru"
+        ];
+        description = ''List of supported languages of Writing Aids (spell checker, grammar checker, thesaurus, hyphenation) on this instance. Allowing too many has negative effect on startup performance.'';
```
```suggestion
        description = ''
          List of supported languages of Writing Aids (spell checker, grammar checker, thesaurus, hyphenation) on this instance.
          Allowing too many has negative effect on startup performance.
        '';
```

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+      # and premium as well. The "base_url" may be https://api.languagetoolplus.com/v2 if the cloud version of LanguageTool is used. Please note that your data in the
+      # document e.g. the text part of it will be sent to the cloud API. Please read the respective privacy policies, e.g. https://languagetool.org/legal/privacy.
+      languagetool = {
+        enabled = lib.mkEnableOption ''Remote Spell and Grammar Checker'';
```
```suggestion
        enabled = lib.mkEnableOption "Remote Spell and Grammar Checker";
```
Please use single quotes for all similar short mkEnableOption and description

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+
+        base_url = lib.mkOption {
+          type = lib.types.str;
+          default = "";
```
```suggestion
          default = "/";
```

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+        };
+
+        user_name = lib.mkOption {
+          type = lib.types.str;
+          default = "";
+          description = ''LanguageTool or Duden account username for premium usage.'';
```
Shouldn't this be by default null?

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+          description = ''LanguageTool or Duden account username for premium usage.'';
+        };
+
+        api_key = lib.mkOption {
```
We cannot provide this securely?

xzfc replied:
I'm not aware if it's possible. Anyways, I've removed all of these options in favor of RFC42-style configuration.

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+          description = ''URL for the API.'';
+        };
+
+        auth_key = lib.mkOption {
```
same here with supplying the credential securely

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+        description = ''Controls whether the systemplate and lotemplate contents are mounted or not, which is much faster than the default of linking/copying each file.'';
+      };
+
+      mount_namespaces = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''Use mount namespaces instead of coolmount.'';
+      };
```
Why not use them?

xzfc replied:
Enabled by default in the updated commit.

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+      server_name = lib.mkOption {
+        type = lib.types.str;
+        default = "";
+        description = ''External hostname:port of the server running coolwsd. If empty, it's derived from the request (please set it if this doesn't work). May be specified when behind a reverse-proxy or when the hostname is not reachable directly.'';
```
```suggestion
        description = ''
          External hostname:port of the server running coolwsd.
          If empty, it's derived from the request (please set it if this doesn't work). 
          May be specified when behind a reverse-proxy or when the hostname is not reachable directly.
        '';
```

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+          };
+
+          anonymization_salt = lib.mkOption {
+            type = lib.types.ints.unsigned; # NOTE: the biggest int Nix accepts is 2^63 - 1
```
Why not use str then?

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+          hosts = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [
+              ''192\.168\.[0-9]{1,3}\.[0-9]{1,3}'' # ............ The IPv4 private 192.168 block as plain IPv4 dotted decimal addresses.
```
Why not use double quotes here?

xzfc replied:
These would require double backslashes.
```nix
''192\.168\.[0-9]{1,3}\.[0-9]{1,3}'' == "192\\.168\\.[0-9]{1,3}\\.[0-9]{1,3}"
"192\.168\.[0-9]{1,3}\.[0-9]{1,3}" == "192.168.[0-9]{1,3}.[0-9]{1,3}"
```

SuperSandro2000 commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+        content_security_policy = lib.mkOption {
+          type = lib.types.listOf lib.types.str;
+          default = [ ];
+          description = ''Customize the CSP header by specifying one or more policy-directive, separated by semicolons. See w3.org/TR/CSP2.'';
```
```suggestion
          description = ''
            Customize the CSP header by specifying one or more policy-directive,
            separated by semicolons. See https://w3.org/TR/CSP2.
          '';
```

SuperSandro2000 reviewed (commented):
We cannot use freeform settings here?

7c6f434c commented:
About configuration: can we just keep the defaults for most stuff, then edit whatever overrides in? And probably it would be done NixOS/RFCs#0042 style with only a few options defined explicitly but whatever the user puts deep into the structure being used for overrides, too…

7c6f434c commented:
Also silly question: is it a good idea to add the package first  and discuss the configuration generation in a follow-up?

drupol commented:
CC @Minion3665

Atemu commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+  ];
+
+  env = {
+    SETCAP = "${libcap}/bin/setcap";
```
Nit:

```suggestion
    SETCAP = lib.getExe' libcap "setcap";
```

xzfc replied:
I've removed this env var as it's not required after bump to `24.04.6-1`.

Atemu commented on pkgs/by-name/co/collabora-online/package.nix:
```diff
@@ -0,0 +1,121 @@
+      substituteInPlace configure.ac --replace-fail '/usr/bin/env python3' python3
+      substituteInPlace coolwsd-systemplate-setup --replace-fail /bin/pwd pwd
+    ''
+    # Use setcap'd wrappers from the wrappers dir, not from the "application path".
+    + ''
+      substituteInPlace common/JailUtil.cpp --replace-fail \
+        'Poco::Path(Util::getApplicationPath(), "coolmount").toString()' \
+        'std::string("/run/wrappers/bin/coolmount")'
+      substituteInPlace wsd/COOLWSD.cpp --replace-fail \
+        ' forKitPath += "coolforkit";' \
+        ' forKitPath = "/run/wrappers/bin/coolforkit";'
+    ''
+    # In the nix build, COOLWSD_VERSION_HASH becomes the same as COOLWSD_VERSION, e.g. `24.04.3.5`.
+    # The web server that serves files from `/browser/$COOLWSD_VERSION_HASH`, doesn't expect the
+    # hash to contain dots.
+    + ''
+      substituteInPlace wsd/FileServer.cpp --replace-fail \
+        'Poco::RegularExpression gitHashRe("/([0-9a-f]+)/");' \
+        'Poco::RegularExpression gitHashRe("/([0-9a-f.]+)/");' \
+    '';
```
These should probably be patches.

7c6f434c replied:
Honestly, at that length of the pattern (spurious hits are not too likely) I'd say either way is fine (LibreOffice packaging uses a mix, because long substitutions are easier to maintain than patches)

xzfc replied:
Removed the first change as it's not required if `mount_namespaces` enabled. (enabled in the NixOS module by default)

Moved the second change to `./fix-file-server-regex.patch`.

Atemu commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,1638 @@
+        RemainAfterExit = true;
+        StateDirectory = "cool";
+        Type = "oneshot";
+        User = "cool";
```
Could this be run using `DynamicUser = true`?

xzfc replied:
`coolwsd-systemplate-setup.service` shares the same data dir as `coolwsd.service`. Yes, it is possible to use `DynamicUser = true`, but that would mean that systemd will move/chown the state dir back and forth.

I.e., after running `coolwsd-systemplate-setup.service`: `/var/lib/cool` is a symlink to `private/cool`, owned by `coolwsd-systemplate`.
Then after running `coolwsd.service`: `/var/lib/cool` is a directory owned by `cool`.

Could you please elaborate on why we might want `DynamicUser` here?

Atemu replied:
If there's state sharing, that's good enough of a reason not to use dynamic user.

Atemu reviewed (commented):
The package diff LGTM apart from a few smaller things.

The module though... It's a beast. I don't think it's reasonable to accept such a huge module.

1. Would it be possible to only pass a subset of options and have the application use the upstream default values on its own?
2. If that is possible, would it be possible to make this a RFC42-style `settings` option so that we don't have to declare any options ourselves?
3. In the docker image, the app takes a [few env vars to configure the most important things](https://sdk.collaboraonline.com/docs/installation/CODE_Docker_image.html). Perhaps we could limit the module to those (via RFC42-style env var option) and then give the user the ability to pass a plain config file text. This would significantly slim the module while still providing useful high-level options to the user.

xzfc commented:
Update:

1. Bump from `24.04.5-3` to `24.04.6-1`. In this version, the new option `mount_namespaces` is introduced. When enabled, the WSD don't call setcap/suid wrappers, so I've removed the related patches and wrappers from the PR.
2. The module is now configured using RFC42-style freeform.
3. Add an update script.

---

I've considered the following alternative freeform approaches:

1. Generate `coolwsd.xml` from scratch. This option assumes that we abandon the default [`coolwsd.xml.in`] and write an almost empty one instead. AFAIU, it's not possible to load the default XML and an additional one. I've abandoned the idea because defaults in `coolwsd.xml.in` are not exactly the same as built-in default values defined in [`COOLWSD.cpp`] despite the comment saying the opposite.

2. Keep the default `coolwsd.xml` and let the user configure command line options through `extraOptions`. E.g. `extraOptions = [ "--o:ssl.enable=false" "--o:ssl.termination=false" ];`. Drawbacks are described in [RFC-42](https://github.com/NixOS/rfcs/blob/62d1245ec1eae275edb996229585637035c02538/rfcs/0042-config-option.md#part-1-structural-settings-instead-of-stringly-extraconfig).

3. Keep the default `coolwsd.xml` and let the user configure options through freeform, and convert it into command line options. It's get a bit complicated when we have to deal with arrays and XML attributes.

4. Implemented in this PR: Merge the values from freeform config into the default [`coolwsd.xml.in`]. It is done by converting `coolwsd.xml` into JSON, then merging it with freeform values using `jq`, then converting it back to XML.

[`coolwsd.xml.in`]: https://github.com/CollaboraOnline/online/blob/cp-24.04.6-3/coolwsd.xml.in
[`COOLWSD.cpp`]: https://github.com/CollaboraOnline/online/blob/cp-24.04.6-3/wsd/COOLWSD.cpp#L2033-L2205

ambroisie commented:
> defaults in coolwsd.xml.in are not exactly the same as built-in default values defined in [COOLWSD.cpp](https://github.com/CollaboraOnline/online/blob/cp-24.04.6-3/wsd/COOLWSD.cpp#L2033-L2205) despite the comment saying the opposite.

This should probably be reported upstream, even if you already implemented a workaround.

ambroisie commented on nixos/modules/services/web-apps/collabora-online.nix:
````diff
@@ -0,0 +1,199 @@
+        <host>::1</host>
+      </post_allow></net>
+      ```
+      in `coolwsd.xml`, or `--o:net.post_allow.host[0]='127\.0\.0\.1 --o:net.post_allow.host=::1` in
````
I'm assuming this should be:

```suggestion
      in `coolwsd.xml`, or `--o:net.post_allow.host[0]='127\.0\.0\.1 --o:net.post_allow.host[1]=::1` in
```

xzfc replied:
Fixed.

ambroisie commented on nixos/modules/services/web-apps/collabora-online.nix:
```diff
@@ -0,0 +1,199 @@
+  options.services.collabora-online = {
+    enable = lib.mkEnableOption "collabora-online";
+
+    package = lib.mkPackageOption pkgs "collabora-online" { };
```
```suggestion
    package = lib.mkPackageOption pkgs "Collaborate Online" { default = "collabora-online"; };
```

xzfc replied:
Fixed.

Minion3665 commented:
> defaults in coolwsd.xml.in are not exactly the same as built-in default values defined in [COOLWSD.cpp](https://github.com/CollaboraOnline/online/blob/cp-24.04.6-3/wsd/COOLWSD.cpp?rgh-link-date=2024-09-09T07%3A15%3A04Z#L2033-L2205) despite the comment saying the opposite.

> This should probably be reported upstream, even if you already implemented a workaround.

(!!) yes, please report this, thanks!

xzfc commented:
> defaults in coolwsd.xml.in are not exactly the same as built-in default values defined in [COOLWSD.cpp](https://github.com/CollaboraOnline/online/blob/cp-24.04.6-3/wsd/COOLWSD.cpp?rgh-link-date=2024-09-09T08%3A54%3A49Z#L2033-L2205) despite the comment saying the opposite.

> This should probably be reported upstream, even if you already implemented a workaround.

Reported upsteam: https://github.com/CollaboraOnline/online/issues/10049.

beardhatcode commented:
@xzfc , thank you for implementing this. I tried to use the alias groups, but it did not work. I just wanted to know if it worked for you. Since from the docs these environment vars should only work for the docker image.

https://sdk.collaboraonline.com/docs/installation/CODE_Docker_image.html?highlight=aliasgroup1

Instead I think this part should actually be filled in:
https://github.com/CollaboraOnline/online/blob/96c54a4e7a066ca8a5787fa310632b8dbb9e6185/coolwsd.xml.in#L269C10-L289

Could you let me know if you are certain the alias groups work in your config, If not I'll likely start drawing up a PR to fill in the config file. :slightly_smiling_face:

Minion3665 commented:
> @/xzfc , thank you for implementing this. I tried to use the alias groups, but it did not work. I just wanted to know if it worked for you. Since from the docs these environment vars should only work for the docker image.

@beardhatcode - The docs are correct in that for regular, manual installations of Collabora Online the alias groups would not work ... however that's controlled by the `--use-env-vars` flag which is present in `ExecStart`, so the environment variables should be respected here.

Relevant PRs are maybe https://github.com/CollaboraOnline/online/pull/7160/files and https://github.com/CollaboraOnline/online/pull/10379
